modules/platforms/cpp/thin-client-test/src/tx_test.cpp

```diff
@@ -0,0 +1,400 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <boost/chrono.hpp>
+#include <boost/thread.hpp>
+#include <boost/test/unit_test.hpp>
+
+#include <ignite/ignition.h>
+
+#include "ignite/thin/ignite_client_configuration.h"
+#include "ignite/thin/ignite_client.h"
+#include "ignite/thin/cache/cache_peek_mode.h"
+#include "ignite/thin/transactions/transaction_consts.h"
+
+#include <test_utils.h>
+#include <ignite/ignite_error.h>
+
+using namespace ignite::thin;
+using namespace boost::unit_test;
+using namespace ignite::thin::transactions;
+using namespace ignite::common::concurrent;
+
+class IgniteTxTestSuiteFixture
+{
+public:
+    IgniteTxTestSuiteFixture()
+    {
+        serverNode = ignite_test::StartCrossPlatformServerNode("cache.xml", "ServerNode");
+    }
+
+    ~IgniteTxTestSuiteFixture()
+    {
+        ignite::Ignition::StopAll(false);
+    }
+
+private:
+    /** Server node. */
+    ignite::Ignite serverNode;
+};
+
+BOOST_FIXTURE_TEST_SUITE(IgniteTxTestSuite, IgniteTxTestSuiteFixture)
+
+bool correctCloseMessage(const ignite::IgniteError& ex)
+{
+    BOOST_CHECK_EQUAL(ex.what(), std::string(TX_ALREADY_CLOSED));
+
+    return true;
+}
+
+bool separateThreadMessage(const ignite::IgniteError& ex)
+{
+    BOOST_CHECK_EQUAL(ex.what(), std::string(TX_DIFFERENT_THREAD));
+
+    return true;
+}
+
+bool checkTxTimeoutMessage(const ignite::IgniteError& ex)
+{
+    return std::string(ex.what()).find("Cache transaction timed out") != std::string::npos;
+}
+
+BOOST_AUTO_TEST_CASE(TestCacheOpsWithTx)
```

Comment:
Let's add a comment with a brief explanation to every test so it would be easier for people with QA background to understand what is going on in test.